base/acme/src/main/java/org/dogtagpki/acme/database/LDAPDatabase.java

```diff
@@ -892,6 +952,69 @@ public void removeExpiredCertificates(Date currentTime) throws Exception {
         }
     }
 
+    public void run() {
+
+        LDAPPersistSearchControl searchControl = new LDAPPersistSearchControl(
+                LDAPPersistSearchControl.MODIFY,
+                false, // return initial entry and subsequent changes
+                true,  // return controls
+                true); // persistent search control is critical
+
+        running = true;
+
+        while (running) {
+            LDAPConnection conn = null;
+            try {
+                conn = connFactory.getConn();
+
+                LDAPSearchConstraints searchConstraints = conn.getSearchConstraints();
+                searchConstraints.setServerControls(searchControl);
+                searchConstraints.setBatchSize(1);
+                searchConstraints.setServerTimeLimit(0);
+
+                logger.info("Start monitoring ACME configuration");
+
+                LDAPSearchResults results = conn.search(
+                    RDN_CONFIG + "," + baseDN,
+                    LDAPConnection.SCOPE_BASE,
+                    "(objectClass=*)",
+                    null,  // return all attributes
+                    false, // return attribute values
+                    searchConstraints);
+
+                while (running && results.hasMoreElements()) {
```

Comment:
Does fetching more elements really take this long that we need to check `running` here? I'd rather complete the current cycle as-is (loading _all_ the config from LDAP) rather than break part-way through and end up with a broken config.

Reply:
The LDAP persistence search will actually keep running until the connection is interrupted. Initially the server will return the search results like in a regular search, but then it will return more results if any of those entries is changed on the server, all in a single persistent search operation.

The `running` flag is generally used to terminate a loop gracefully in a thread (instead of using `Thread.stop()`), so the loop will finish the current iteration but it will not execute the next one.

Reply:
Ah I see it now. Thanks for explaining on IRC :)